frame/ethereum/src/lib.rs

```diff
@@ -308,8 +308,7 @@ impl<T: Config> Module<T> {
 			mix_hash: H256::default(),
 			nonce: H64::default(),
 		};
-		let mut block = ethereum::Block::new(partial_header, transactions.clone(), ommers);
-		block.header.state_root = T::StateRoot::get();
+		let block = ethereum::Block::new(partial_header, transactions.clone(), ommers);
```

Comment:
It would be better to pass `receipts`  to `ethereum::Block::new()` just like `transactions`,  but we need to update ethereum crate, this may take a longer time. This depends on feedback from the reviewer, if necessary, I will do it.